test/NuGet.Core.Tests/NuGet.Packaging.Test/ReferencesInNuspecMatchRefAssetsRuleTests.cs

```diff
@@ -0,0 +1,280 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Castle.DynamicProxy.Generators.Emitters.SimpleAST;
+using NuGet.Common;
+using NuGet.Packaging.Rules;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class ReferencesInNuspecMatchRefAssetsRuleTests
+    {
+        [Fact]
+        public void Compare_PackageWithReferencesAndRefFolderMatching_ShouldBeEmpty()
+        {
+            //Arrange
+            var references = new Dictionary<string, string[]>
+            {
+                {"net462", new string[] {"MyLib.dll", "MyHelpers.dll"} }
+            };
+            var files = new string[]
+            {
+                "ref/net462/MyLib.dll",
+                "ref/net462/MyHelpers.dll"
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingItems = rule.Compare(references, files);
+
+            //Assert
+            Assert.Empty(missingItems);
+        }
+
+        [Fact]
+        public void Compare_PackageWithReferencesThatTheRefFolderDoesNotHave_RefFilesMissingShouldNotBeEmpty()
+        {
+            //Arrange
+            var references = new Dictionary<string, string[]>
+            {
+                {"net462", new string[] {"MyLib.dll", "MyHelpers.dll"} }
+            };
+            var files = new string[]
+            {
+                "ref/net462/MyLib.dll",
+            };
+
+            //Act
+            var rule = new ReferencesInNuspecMatchRefAssetsRule();
+            var missingItems = rule.Compare(references, files);
+
+            //Assert
+            Assert.Equal(missingItems.Count(), 1);
+            var singleIssue = missingItems.Single(t => t.TFM.Equals("net462") && t.MissingItems.Contains("MyHelpers.dll"));
```

Comment:
If this test fails, it will give an error message that is difficult to understand. Also, you don't validate that `MissingItems` contains the expected count of items, so there can be false positives.
```csharp
Assert.Equal("net462", missingItems[0].Tfm);
Assert.Equal(1, missingItems[0].MissingItems);
Assert.Equal("MyHelpers.dll", missingItems[0].MissingItems[0]);
```

reduces the risk of false positive results.